vllm/attention/selector.py

```diff
@@ -226,7 +227,10 @@ def which_attn_to_use(
         elif kv_cache_dtype is not None and kv_cache_dtype.startswith("fp8"):
             logger.info(
                 "Cannot use FlashAttention-2 backend for FP8 KV cache.")
-            selected_backend = _Backend.XFORMERS
+            if is_flashinfer():
+                selected_backend = _Backend.FLASHINFER
+            else:
+                selected_backend = _Backend.XFORMERS
```

Comment:
Again since FlashInfer is one of attention backends, I'd suggest let users explicitly specify, so the logic would be:
1. If user specifies FlashAttn+fp8 kv cache, fallback to xformers with warning saying that FlashInfer is recommended.
2. If user specifies FlashInfer+fp8 kv cache, use FlashInfer.

Also since FlashInfer fp8 doesn't support bfloat16, you need to check whether the model dtype is float16; otherwise we should fallback to xformers.

Reply:
The intention of the PR is to opt into higher throughput when available. What disadvantage do you see if this path is enabled without user intervention?

Reply:
@comaniac Alternately, do you know when FlashInfer backend can be made default instead of FlashAttention?

Reply:
I don't think it would be a case soon, majority because we haven't conducted a comprehensive test and benchmark to guarantee the performance and result quality.

Due to the same reason, the current use of FlashInfer is all by explicit enabling. For example we also have USE_FLASHINFER_SAMPLER for now.

Reply:
Unlike `USE_FLASHINFER_SAMPLER`, the scope of this path is very limited. I discussed with @simon-mo: do you have any further inputs?

Reply:
Both of you have a point. @pavanimajety can you comment on the test and benchmark for the quality of this kernel?   and the bfloat16 issue (i think that's an edge case we need to cover) think we can default to this if the quality is good

Reply:
I was able to test for this specification and `kv_cache_dtype=fp8` with `torch.float8_e4m3fn`. works in all these cases:
```
@pytest.mark.parametrize("kv_lens", [[1328, 18, 463], [1, 54, 293, 70]])
@pytest.mark.parametrize("num_heads", [(32, 8), (64, 8), (6, 1)])
@pytest.mark.parametrize("head_size", [128, 256])
@pytest.mark.parametrize("block_size", [16,32])
@pytest.mark.parametrize("dtype", [torch.bfloat16, torch.float16])
@pytest.mark.parametrize("soft_cap", [None, 30.0, 50.0])
@torch.inference_mode
```

Please note: num_query_heads=16 and num_kv_heads=16 case didn't work. I can add a To Do for follow up with that case. 

Did you want to see any more benchmarks outside of what I have in the main PR comment?